src/main/java/org/toradocu/extractor/JavadocExtractor.java

```diff
@@ -1,515 +1,664 @@
+   * @throws ClassNotFoundException if a type described in a Javadoc comment cannot be loaded (e.g.,
+   *     the type is not on the classpath)
    */
-  public List<DocumentedMethod> extract(ClassDoc classDoc) throws IOException {
-    List<DocumentedMethod> methods = new ArrayList<>();
-
-    // Loop on constructors and methods (also inherited) of the target class.
-    for (ExecutableMemberDoc member : getConstructorsAndMethods(classDoc)) {
-      ClassDoc containingClass = member.containingClass();
-      String containingClassName = containingClass.name();
-      if (containingClassName.contains(".")) { // Containing class is not a top-level class.
-        containingClassName = containingClassName.replace(".", "$");
-      }
-      List<ThrowsTag> memberThrowsTags = extractThrowsTags(member, classDoc);
-      List<ParamTag> memberParamTags = extractParamTags(member, classDoc);
-      List<ReturnTag> memberReturnTags = extractReturnTag(member, classDoc);
-
-      ReturnTag finalReturnTag = null;
-      if (!memberReturnTags.isEmpty()) {
-        finalReturnTag = memberReturnTags.get(0);
+  private BlockTags createTags(
+      List<String> classesInPackage,
+      CallableDeclaration<?> sourceMember,
+      List<DocumentedParameter> parameters,
+      String className)
+      throws ClassNotFoundException {
+
+    List<ParamTag> paramTags = new ArrayList<>();
+    ReturnTag returnTag = null;
+    List<ThrowsTag> throwsTags = new ArrayList<>();
+
+    final Optional<Javadoc> javadocOpt = sourceMember.getJavadoc();
+    if (javadocOpt.isPresent()) {
+      final Javadoc javadocComment = javadocOpt.get();
+      final List<JavadocBlockTag> blockTags = javadocComment.getBlockTags();
+      for (JavadocBlockTag blockTag : blockTags) {
+        switch (blockTag.getType()) {
+          case PARAM:
+            ParamTag paramTag = createParamTag(blockTag, parameters);
+            if (paramTag != null) {
+              // createParamTag returns null in case of generic type.
+              paramTags.add(paramTag);
+            }
+            break;
+          case RETURN:
+            returnTag = createReturnTag(blockTag);
+            break;
+          case EXCEPTION:
+          case THROWS:
+            throwsTags.add(createThrowsTag(classesInPackage, blockTag, sourceMember, className));
+            break;
+          default:
+            // ignore other block tags
+            break;
+        }
       }
+    }
+    return new BlockTags(paramTags, returnTag, throwsTags);
+  }
 
-      methods.add(
-          new DocumentedMethod(
-              new org.toradocu.extractor.Type(
-                  containingClass.containingPackage() + "." + containingClassName),
-              member.name(),
-              getReturnType(member),
-              getParameters(member),
-              memberParamTags,
-              member.isVarArgs(),
-              memberThrowsTags,
-              finalReturnTag));
+  /**
+   * Instantiate a tag of throws kind.
```

Comment:
It's confusing that the Javadoc says "instantiate" but the method name is "create".  Stick with consistent terminology.